### PR TITLE
fix: global button comp to woo commerce button.

### PR DESCRIPTION
**Pull request by shubhw12**
### Description

Added the global button border support to the woocommerce buttons as well.
### Screenshots

https://share.getcloudapp.com/xQuDw1yZ

### Types of changes

Bug fix.

### How has this been tested?

I have tested with backward compatibility

### Checklist:
- [x] My code is tested
- [x] My code passes the PHPCS tests
- [x] My code follows accessibility standards 
- [x] My code has proper inline documentation 
- [x] I've included any necessary tests 
- [x] I've included developer documentation 
- [x] I've added proper labels to this pull request

### CODE REVIEW

File: inc/compatibility/woocommerce/class-astra-woocommerce.php [hunk 1]
```diff
@@ -1236,6 +1241,35 @@ public function add_styles() {
...
+			/*
+			* global button settings not working for woocommerce button on shop and single page.
+			* check if the current user is existing user or new user.
+			* if new user load the CSS bty default if existing provide a filter
+			*/
+			if ( self::astra_gbl_btn_woo_comp() ) {
+
+				$woo_global_button_css = array(
+					'.woocommerce a.button , .woocommerce button.button.alt' => array(
+						'border' => '0',
```
Comment (imnavanath) [inline inc/compatibility/woocommerce/class-astra-woocommerce.php:1253]: No need to move this static CSS into dynamic.

File: inc/compatibility/woocommerce/class-astra-woocommerce.php [hunk 2]
```diff
@@ -1477,6 +1511,18 @@ public function cart_link_fragment( $fragments ) {
...
+		/**
+		 * For existing users, do not load the wide/full width image CSS by default.
+		 *
+		 * @since x.x.x
+		 * @return boolean false if it is an existing user , true if not.
+		 */
+		public static function astra_gbl_btn_woo_comp() {
+			$astra_settings                          = get_option( ASTRA_THEME_SETTINGS );
+			$astra_settings['astra_gbl_btn_woo_css'] = isset( $astra_settings['astra_gbl_btn_woo_css'] ) ? false : true;
+			return apply_filters( 'astra_gbl_btn_woo_comp', $astra_settings['astra_gbl_btn_woo_css'] );
```
Comment (imnavanath) [inline inc/compatibility/woocommerce/class-astra-woocommerce.php:1523]: @shubhw12 can you please update the Filter name.

File: inc/theme-update/astra-update-functions.php [hunk 3]
```diff
@@ -219,3 +219,20 @@ function astra_gtn_full_wide_image_group_css() {
...
+ * @since 2.5.0
+ *
+ * @return void
+ */
+function astra_global_button_woo_css() {
+	$theme_options = get_option( 'astra-settings', array() );
+
+	// Set flag to not load button specific CSS.
+	if ( ! isset( $theme_options['astra_gbl_btn_woo_css'] ) ) {
+		$theme_options['astra_gbl_btn_woo_css'] = false;
```
Comment (imnavanath) [inline inc/theme-update/astra-update-functions.php:235]: Filter name can come here as well, for this option.

File: inc/compatibility/woocommerce/class-astra-woocommerce.php (hunk 1 above)
Comment (shubhw12) [reply inc/compatibility/woocommerce/class-astra-woocommerce.php]: Done

File: inc/compatibility/woocommerce/class-astra-woocommerce.php (hunk 2 above)
Comment (shubhw12) [reply inc/compatibility/woocommerce/class-astra-woocommerce.php]: Done

File: inc/theme-update/astra-update-functions.php (hunk 3 above)
Comment (shubhw12) [reply inc/theme-update/astra-update-functions.php]: Done

File: inc/theme-update/astra-update-functions.php [hunk 4]
```diff
@@ -219,3 +219,20 @@ function astra_gtn_full_wide_image_group_css() {
...
+ *
+ * @since x.x.x
+ *
+ * @return void
+ */
+function astra_global_button_woo_css() {
+	$theme_options = get_option( 'astra-settings', array() );
+
+	// Set flag to not load button specific CSS.
+	if ( ! isset( $theme_options['astra-global-btn-woo-css'] ) ) {
```
Comment (imnavanath) [inline inc/theme-update/astra-update-functions.php:234]: Can we add `class_exists( 'WooCommerce' )` check here, so the new flag will only set for those users, who used WooCommerce on their site?

Comment (shubhw12) [reply inc/theme-update/astra-update-functions.php:234]: As of now, we can proceed with the existing method. we can set all the backward tasks on a single task for this.

File: inc/theme-update/class-astra-theme-background-updater.php [hunk 5]
```diff
@@ -53,7 +53,7 @@ class Astra_Theme_Background_Updater {
 				'astra_gtn_full_wide_image_group_css',
 			),
 			'2.5.0' => array(
-				'astra_gtn_full_wide_group_cover_css',
```
Comment (premanshup) [inline inc/theme-update/class-astra-theme-background-updater.php:56]: @shubhw12 Please check this, `astra_gtn_full_wide_group_cover_css` this function should not be removed. Please compare it with `next-release` once.

File: inc/theme-update/astra-update-functions.php [hunk 6]
```diff
@@ -221,18 +221,18 @@ function astra_gtn_full_wide_image_group_css() {
...
  * @since x.x.x
  *
  * @return void
  */
-function astra_gtn_full_wide_group_cover_css() {
-
+function astra_global_button_woo_css() {
 	$theme_options = get_option( 'astra-settings', array() );
 
-	if ( ! isset( $theme_options['gtn-full-wide-grp-cover-css'] ) ) {
```
Comment (premanshup) [inline inc/theme-update/astra-update-functions.php:234]: @shubhw12 Please check this, 'gtn-full-wide-grp-cover-css' this comp should not be removed. Please compare it with next-release once.

**Pull request closed without merging**